### ISSUE TITLE
Loading PDF Page from pdf prop

### ISSUE THREAD

**Issue by Rioos** [wojtekmaj/react-pdf#201]
I'm using version 3.0.4 and I was trying to load a pdf page from the pdf property and found out that the code isn't supporting this anymore. It only tries to load the PDF from the "context" object and it completely ignores "props" object.

Is that the expected behavior? If so, you need to fix docs on 3.0.4 that says that's possible.

If not I just forked the project and can PR the fix so you can update that on NPM, that would be perfect.

I saw that on 4.x version it seems to be solved as well (didn't tested tho).

```js

// @Page.jsx

  // Version 3.0.4
  loadPage(props = this.props, context = this.context) {
    const { pdf } = context; 

 // Proposed fix for 3.0.5
  loadPage(props = this.props, context = this.context) {
    const pdf = props.pdf || context.pdf;

 //Code on master
  loadPage = async () => {
    const { pdf } = this.props;

```

**Comment by wojtekmaj** [comment]
Yes, it is expected. You should wrap `<Page>` components with `<Document>` component which will load the file for you. If you try to use PDF.js to handle loading PDF on your own, many options will not work, like interactive annotations, links, ToC etc. Please refer to basic example of usage in README for proper implementation.

**Comment by Rioos** [comment]
On [v3.0.4 docs](https://github.com/wojtekmaj/react-pdf/tree/v3.0.4#page ) you're saying that it is possible:

> Displays a page. Must be placed inside <Document /> **or have pdf prop passed, which can be obtained from <Document />'s onLoadSuccess callback function.**

Right after that you affirm that:

> Note: <Page/> must be a direct child of <Document /> component. <Document /> passes necessary props only to its direct children. If you wish to put a component in between of <Document /> and <Page/>, you must ensure to pass all the props to <Page/> component by yourself.

So I'm not really sure on what to expect. But it definitely says that it is a possibility.

The possibility of having a page outside a Document tag is awesome, even without the features you're saying.

**Comment by wojtekmaj** [comment]
Ah! You're absolutely correct, that's a bug. Fixed already in 4.x branch by the way :)

**Comment by wojtekmaj** [comment]
This should be fixed now in v3.0.5. Please update and let me know of the outcome!

**Comment by Rioos** [comment]
Working perfect!
